### ISSUE TITLE
Boson: Disable all Non-Uniformity Correction

### ISSUE THREAD

**Issue by vishwa91** [LJMUAstroecology/flirpy#46]
First, thanks a ton for the wonderful codebase -- made my life so much easier.

I wanted to know if it is possible to disable all corrections on the Boson 640 camera? I want to get raw data for performing custom NUC and hence the question.

Thank you.

**Comment by jveitchmichaelis** [comment]
Hiya, this is probably possible. You'd likely need multiple commands because there are several stages of correction that the camera performs by default (and I'm not sure which you'd want to turn off for your application). It's a matter of implementing the commands defined in the Boson Software idd, but that's fairly easy if you know which. We'd need to add the relevant commands to the Boson class - ultimately it'd be good to implement the entire camera interface at some point...

**Comment by vishwa91** [comment]
I agree with the multi-step command, as the GUI requires switching off several options. I will look into the Boson commands reference and try them to this codebase. Thanks!

**Comment by jveitchmichaelis** [comment]
Yes, each command should be a separate API call. If you get them working, please consider submitting a pull request (ideally also with a test case for each command- so you should write a set/getfunction for each). Otherwise if you have any questions about implementation, feel free to ask.

**Comment by vishwa91** [comment]
I will follow the same template as the code base, and add test functions. Once I test on the Boson camera I have, I will submit a pull request. Thanks for the support!

**Comment by nrwiederhold** [comment]
What is the status of this issue? Do these options exist in the code base now?

**Comment by jveitchmichaelis** [comment]
No PR was submitted, but you're welcome to add it - it should be quite easy to do, but I don't have access to a Boson at the moment.

**Comment by socome** [comment]
@jveitchmichaelis Could you teach me how to turn off the NUC on boson camera .? I will try and PR about that. I want to turn off the NUC because of image block when NUC is running

**Comment by jveitchmichaelis** [comment]
Do you mean flat fielding? This causes a slight pause when the camera takes a calibration image. This is different to NUC.

**Comment by socome** [comment]
@jveitchmichaelis Oh .. When i read [this post](https://www.flir.com/discover/professional-tools/what-is-a-non-uniformity-correction-nuc/), i think NUC is main reason of thermal image sometimes freezes and the camera makes a clicking noise. However, do you mean this problem caused by FFC..? I will try to use the "set_ffc_manual" function. Thank you!

**Comment by jveitchmichaelis** [comment]
Ah I think they use it interchangeably. From the article:

>  It is also referred to as an FFC (flat field correction

Usually I’ve seen NUC used to refer to the various other settings the camera uses to correct the image (see the beginning of this thread).
